src/probnum/pnm/_stopping_criterion.py

```diff
@@ -0,0 +1,92 @@
+"""Stopping criterion for a (probabilistic) numerical method."""
+
+import abc
+
+
+class StoppingCriterion(abc.ABC):
+    r"""Stopping criterion of a (probabilistic) numerical method.
```

Comment:
why is probabilistic in brackets? I suppose you mean that some of the stopping criteria can also be used by non-probabilistic methods. But since ProbNum only contains PNMs, we can drop the brackets here.